server_addon/maya/client/ayon_maya/api/lib.py

```diff
@@ -1734,6 +1734,15 @@ def is_valid_reference_node(reference_node):
     sel = OpenMaya.MSelectionList()
     sel.add(reference_node)
     depend_node = sel.getDependNode(0)
+    # maya 2022 is missing `isValidReference` so the check needs to be
+    # done in different way.
+    if cmds.about(version=True) < 2023:
+        try:
+            cmds.referenceQuery(reference_node, filename=True)
+            return True
+        except RuntimeError:
+            return False
```

Comment:
Technically this can move even above the `sel = OpenMaya.MSelectionList` line.
```suggestion
    # maya 2022 is missing `isValidReference` so the check needs to be
    # done in different way.
    if cmds.about(version=True) < 2023:
        try:
            cmds.referenceQuery(reference_node, filename=True)
            return True
        except RuntimeError:
            return False
    sel = OpenMaya.MSelectionList()
    sel.add(reference_node)
    depend_node = sel.getDependNode(0)
```